hibernate-core/src/main/java/org/hibernate/persister/collection/BasicCollectionPersister.java

```diff
@@ -512,7 +512,7 @@ public static class MyEmbeddable {
 
 			 	For this reason we cannot take into consideration the `@Column(updatable = false)`
 			 */
-			attribute.getElementDescriptor().forEachNonFormula( updateBuilder );
+			attribute.getElementDescriptor().forEachInsertable( updateBuilder );
```

Comment:
Why `forEachInsertable` and not `forEachUpdatable`? This is the `generateUpdateRowAst` method after all, so IMO if this must use `forEachInsertable`, there should be a comment explaining the need for this.

Reply:
The comment is already there, we were already in a special case that included non-updatable columns here by using `forEachNonFormula`.

Reply:
Well the comment is already there, this was already a special case for indexed-element collections that ignored `updatable=false` since we were using `forEachNonFormula`.